lib/cloudflare/rails/railtie.rb

```diff
@@ -79,6 +79,14 @@ def fetch_with_cache(type)
               send type
             end
           end
+
+          def cloudflare_ips(refresh: false)
+            @ips = nil if refresh
```

Comment:
i'm concerned about thread safety here. what happens if multiple imports run at the same time? previously this was less likely because it would require multiple rails instances coming up simultaneously, but it's certainly possible for multiple threads. 

maybe a semaphore would help, or something like `Concurrent::Array`? i could also be over thinking this.

Reply:
I don’t think it would be a problem, worst case the last import would win. Generally it would not matter, as the value should be the same, but to be extra safe it could be cached in the thread local storage instead of in an instance variable.

Reply:
Although actually a `Mutex` is not needed here: setting an instance variable is thread safe, even a class instance variable like in this case. The instance variable is set to an array, and the array is never mutated (it’s a good idea to freeze it), so there is no need for a `Concurrent::Array`.